src/main/java/org/sciborgs1155/robot/pivot/Pivot.java

```diff
@@ -0,0 +1,160 @@
+package org.sciborgs1155.robot.pivot;
+
+import static edu.wpi.first.units.Units.*;
+import static org.sciborgs1155.robot.pivot.PivotConstants.*;
+
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine.Direction;
+import java.util.function.Supplier;
+import monologue.Annotations.Log;
+import monologue.Logged;
+import org.sciborgs1155.robot.Constants;
+import org.sciborgs1155.robot.Robot;
+
+public class Pivot extends SubsystemBase implements AutoCloseable, Logged {
+  @Log.NT private final PivotIO pivot;
+  private final SysIdRoutine sysIdRoutine; //sysIdRoutineoogabooga
+
+  // pivot control
+  @Log.NT 
+  final ProfiledPIDController pivotPID =
+      new ProfiledPIDController(
+          PivotConstants.kP,
+          PivotConstants.kI,
+          PivotConstants.kD,
+          new TrapezoidProfile.Constraints(PivotConstants.MAX_VELOCITY, PivotConstants.MAX_ACCEL));
+  @Log.NT private final ArmFeedforward pivotFeedforward =
+      new ArmFeedforward(PivotConstants.kS, PivotConstants.kG, PivotConstants.kV);
+
+  // climb control
+  @Log.NT private final ProfiledPIDController climbPID =
+      new ProfiledPIDController(
+          ClimbConstants.kP,
+          ClimbConstants.kI,
+          ClimbConstants.kD,
+          new TrapezoidProfile.Constraints(ClimbConstants.MAX_VELOCITY, ClimbConstants.MAX_ACCEL));
+  @Log.NT private final ArmFeedforward climbFeedforward =
+      new ArmFeedforward(ClimbConstants.kS, ClimbConstants.kG, ClimbConstants.kV);
+
+  // visualization
+  @Log.NT final Mechanism2d measurement = new Mechanism2d(3, 4);
+  @Log.NT final Mechanism2d setpoint = new Mechanism2d(3, 4);
+
+  private final PivotVisualizer positionVisualizer =
+      new PivotVisualizer(measurement, new Color8Bit(255, 0, 0));
+  private final PivotVisualizer setpointVisualizer =
+      new PivotVisualizer(setpoint, new Color8Bit(0, 0, 255));
+
+  /** Creates a real or simulated pivot based on {@link Robot#isReal()} */
+  public static Pivot create() {
+    return Robot.isReal() ? new Pivot(new RealPivot()) : new Pivot(new SimPivot());
+  }
+
+  public Pivot(PivotIO pivot) {
+    this.pivot = pivot;
+    sysIdRoutine =
+        new SysIdRoutine(
+            new SysIdRoutine.Config(),
+            new SysIdRoutine.Mechanism(
+                (v) -> pivot.setVoltage(v.in(Volts)), null, this, "pivotSysId"));
```

Comment:
instead of `null` here, we need to actually log our position and velocity from our encoder
to do this, provide a lambda that takes in log and logs our motor
something like this should work:
```java
log.motor("pivot").voltage(pivot.getAppliedVoltage()).angularPosition(pivot.getPosition().toRadians()).angularVelocity(pivot.getVelocity())
```
although i'm not entirely sure if we need the voltage, urcl should get that for us

also, we don't need to supply "pivotSysId" as the name, since sysid will use the class name "Pivot" by default

Reply:
Jank solution for velocity, should work though. Prob should be noted.